hub/core/storage/lru_cache.py

```diff
@@ -107,6 +105,29 @@ def __delitem__(self, path: str):
             if not deleted_from_cache:
                 raise
 
+    def clear_cache(self):
+        """Flushes the content of the cache and and then deletes contents of all the layers of it.
+        This doesn't delete data from the actual storage.
+        """
+        self.flush()
+        self.cache_used = 0
+        self.lru_sizes.clear()
+        self.dirty_keys.clear()
+        self.cache_storage.clear()
+
+        if self.next_storage.hasattr("clear_cache"):
+            self.next_storage.clear_cache()
+
+    def clear(self):
```

Comment:
This should probably not delete data from the last storage but clear the cache only. A side-effect of this would be actually deletion of datasets where this looks like to a user that the only cache is being cleared.

Or we rename this to "delete".

Reply:
As I mentioned earlier this function deletes everything. It's a function that the cache implements by itself using iter and delitem. The only reason to override this was to make it faster using storage specific .clear methods that are called within it.